packages/website/hooks/use-payment.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+
+import { userBillingSettings } from '../lib/api';
+import { plansAll } from '../components/contexts/plansContext';
+
+export const usePayment = () => {
+  /**
+   * @typedef {Object} Plan
+   * @property {string | null} id
+   * @property {string} title
+   * @property {string} description
+   * @property {string} price
+   * @property {string} base_storage
+   * @property {string} additional_storage
+   * @property {string} bandwidth
+   * @property {string} block_limit
+   * @property {string} car_size_limit
+   * @property {boolean} pinning_api
+   */
+
+  const [hasPaymentMethod, setHasPaymentMethod] = useState(false);
+  const [currentPlan, setCurrentPlan] = useState(/** @type {Plan | null} */ (null));
+
+  useEffect(() => {
+    const getUserPaymentInfo = async () => {
+      const userPlan = await userBillingSettings();
+      console.log(userPlan);
+      if (userPlan.paymentMethod) {
+        await setHasPaymentMethod(true);
```

Comment:
I'm not sure we really need state that just stores a boolean and 'has'. You can have state for the whole fetched paymentSettings, and you can derrive (`useMemo`) paymentMethod from that, and you can derrive `hasPaymentMethod` from that.

I find that `useMemo` is easier to reason about and maintain than `useEffect`, so try to use it whenever possible.

Reply:
Regarding this hook, it was previously merged code that I was just drying up, but I have updated it to be inline with your new approach on payment.js, ultimately it could all come from this hook maybe? but for now I'm not going to touch payment.js